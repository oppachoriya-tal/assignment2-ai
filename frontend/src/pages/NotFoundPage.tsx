import React from 'react';
import { Container, Typography, Box } from '@mui/material';

const NotFoundPage: React.FC = () => {
  return (
    <Container maxWidth="sm" data-testid="not-found-page">
      <Box sx={{ py: 4, textAlign: 'center' }}>
        <Typography variant="h1" component="h1" gutterBottom>
          404
        </Typography>
        <Typography variant="h4" component="h2" gutterBottom>
          Page Not Found
        </Typography>
        <Typography variant="body1" color="text.secondary">
          The page you are looking for does not exist.
        </Typography>
      </Box>
    </Container>
  );
};

export default NotFoundPage;
