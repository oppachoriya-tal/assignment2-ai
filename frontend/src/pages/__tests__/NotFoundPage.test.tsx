import React from 'react';
import { render, screen } from '@testing-library/react';
import NotFoundPage from '../NotFoundPage';

describe('NotFoundPage Component', () => {
  it('renders 404 page with correct content', () => {
    render(<NotFoundPage />);
    
    expect(screen.getByTestId('not-found-page')).toBeInTheDocument();
    expect(screen.getByText('404')).toBeInTheDocument();
    expect(screen.getByText('Page Not Found')).toBeInTheDocument();
    expect(screen.getByText('The page you are looking for does not exist.')).toBeInTheDocument();
  });

  it('has correct heading structure', () => {
    render(<NotFoundPage />);
    
    const h1 = screen.getByRole('heading', { level: 1 });
    const h2 = screen.getByRole('heading', { level: 2 });
    
    expect(h1).toHaveTextContent('404');
    expect(h2).toHaveTextContent('Page Not Found');
  });

  it('has correct container structure', () => {
    render(<NotFoundPage />);
    
    const container = screen.getByTestId('not-found-page');
    expect(container).toBeInTheDocument();
    expect(container).toHaveClass('MuiContainer-root');
  });

  it('renders consistently', () => {
    const { container } = render(<NotFoundPage />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it('has accessible text content', () => {
    render(<NotFoundPage />);
    
    // Check that headings are accessible
    expect(screen.getByRole('heading', { level: 1 })).toBeInTheDocument();
    expect(screen.getByRole('heading', { level: 2 })).toBeInTheDocument();
    
    // Check that descriptive text is accessible
    const description = screen.getByText('The page you are looking for does not exist.');
    expect(description).toBeInTheDocument();
  });

  it('has proper semantic structure', () => {
    render(<NotFoundPage />);
    
    // Should have proper heading hierarchy
    const headings = screen.getAllByRole('heading');
    expect(headings).toHaveLength(2);
    expect(headings[0]).toHaveTextContent('404');
    expect(headings[1]).toHaveTextContent('Page Not Found');
  });
});
