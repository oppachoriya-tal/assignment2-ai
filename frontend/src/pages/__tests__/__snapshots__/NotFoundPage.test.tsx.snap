// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage Component renders consistently 1`] = `
<div
  class="MuiContainer-root MuiContainer-maxWidthSm css-cuefkz-MuiContainer-root"
  data-testid="not-found-page"
>
  <div
    class="MuiBox-root css-1aicb05"
  >
    <h1
      class="MuiTypography-root MuiTypography-h1 MuiTypography-gutterBottom css-a4d0r7-MuiTypography-root"
    >
      404
    </h1>
    <h2
      class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
    >
      Page Not Found
    </h2>
    <p
      class="MuiTypography-root MuiTypography-body1 css-1pnmrwp-MuiTypography-root"
    >
      The page you are looking for does not exist.
    </p>
  </div>
</div>
`;
